src/main/java/seedu/address/logic/commands/ListRecordCommand.java

```diff
@@ -0,0 +1,83 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_RECORDS;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Lists all records for a specific patient to the user.
+ */
+public class ListRecordCommand extends Command {
+
+    public static final String COMMAND_WORD = "listR";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Shows a list of all records for the specific patient in the records database.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_SUCCESS = "Listed records for this patient: ";
+
+    private static Person lastCalledPerson;
+
+    private final Index targetIndex;
+
+    public ListRecordCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    /**
+     * Checks if listR command is previously called.
+     *
+     * @return true if listR command is previously called, false otherwise.
+     */
+    public static boolean isCalled() {
+        return ListRecordCommand.lastCalledPerson != null;
```

Comment:
I think this implementation might run into an issue, when we perfom a series of steps like:
1) Call `listR 1` [lastCalledPerson = Person at index 1, RecordList shown]
2) Call `list`  [Address Book is shown]
3) Call `addR` [lastCalledPerson != null, #`isCalled` returns true]

Might need to reset lastCalledPerson when non-record commands are called

Reply:
True, will definitely present a problem when non-Record commands are called after the `listR` command, if must strictly follow the rule that `deleteR` and `findR` can only be called immediately after `listR` is called.